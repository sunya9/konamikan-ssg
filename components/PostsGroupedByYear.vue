<template>
  <ul>
    <li v-for="post in posts" :key="post.id">
      <div class="oneline">
        <nuxt-link :to="$resolvePostUrl(post)">
          <span class="has-text-grey">
            {{ $dayjs(post.published_at).format('MM/DD') }}
          </span>
          <span>{{ post.title }}</span>
          <span v-if="post.excerpt" class="has-text-grey excerpt">
            {{ post.custom_excerpt || post.excerpt }}
          </span>
        </nuxt-link>
      </div>
    </li>
  </ul>
</template>
<script lang="ts">
import Vue from 'vue'
import { Component, Prop } from 'nuxt-property-decorator'
import { PostOrPage } from '@tryghost/content-api'

@Component({})
export default class PostsGroupedByYear extends Vue {
  @Prop({ type: Array, required: true })
  posts!: PostOrPage[]

  @Prop({ type: Number, required: true })
  year!: number
}
</script>
<style lang="scss" scoped>
@import '~bulma/sass/utilities/all';

.oneline {
  @include tablet {
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  }
  .excerpt {
    display: none;
    @include tablet {
      display: inline;
    }
  }
}
</style>
