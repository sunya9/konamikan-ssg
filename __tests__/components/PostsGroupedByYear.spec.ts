import { mount } from '@vue/test-utils'
import { post } from '@/__tests__/fixtures/postOrPage'
import PostsGroupedByYear from '@/components/PostsGroupedByYear.vue'

describe('PostsGroupedByYear', () => {
  it('normal', () => {
    expect(
      mount(PostsGroupedByYear, {
        propsData: {
          posts: [post],
          year: 2020
        }
      })
    ).toMatchSnapshot()
  })
})
