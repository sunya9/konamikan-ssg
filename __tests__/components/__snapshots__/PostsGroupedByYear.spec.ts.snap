// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsGroupedByYear normal 1`] = `
<ul>
  <li>
    <div class="oneline"><a><span class="has-text-grey">
          04/05
        </span> <span>title</span> <span class="has-text-grey excerpt">
          excerpt
        </span></a></div>
  </li>
</ul>
`;
